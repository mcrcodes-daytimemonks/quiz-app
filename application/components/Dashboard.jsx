import PropTypes from "prop-types";
import { useEffect, useState } from "react";
import { useRouter } from "next/router";
const categories = ["JavaScript", "HTML", "mySQL"];

const Dashboard = ({ handleLogout }) => {
  const [questions, setQuestions] = useState([]);
  const [questionLimit, setQuestionLimit] = useState(1);
  const [username, setUsername] = useState("");
  const router = useRouter();

  const goToGame = () => router.push("/game");

  const returnToGameOptions = () => {
    ["questions", "questionIndex", "category"].forEach((key) =>
      localStorage.setItem(key, "")
    );
    router.reload();
  };

  const storeCategory = (category) =>
    localStorage.setItem("category", category);

  const storeQuestionLimit = (event) => {
    localStorage.setItem("questionLimit", event.target.value);
    setQuestionLimit(Number(event.target.value));
  };

  useEffect(() => {
    const storedQuestions = localStorage.getItem("questions");
    if (storedQuestions) {
      setQuestions(JSON.parse(storedQuestions));
    }
    setUsername(localStorage.getItem("username"));
  }, []);

  return (
    <div>
      <h1>Dashboard</h1>
      {questions.length ? (
        <div>
          <p>
            You are currently half way through a previous game. What would you
            like to do?
          </p>
          <button type="button" onClick={goToGame}>
            Continue
          </button>
          <button type="button" onClick={returnToGameOptions}>
            New Game
          </button>
        </div>
      ) : (
        <div>
          <p>
            Welcome, <span>{username}</span>
          </p>
          <p>Please choose a category</p>
          {categories.map((category, i) => (
            <button key={category} onClick={() => storeCategory(category)}>
              {category}
            </button>
          ))}
          <label htmlFor="question-limit">How many questions?</label>
          <input
            type="number"
            step="1"
            min="1"
            onChange={storeQuestionLimit}
            value={questionLimit}
          />
          <br />
          <button type="button" onClick={goToGame}>
            Play
          </button>
          <br />
          <button type="button" onClick={handleLogout}>
            Logout
          </button>
        </div>
      )}
    </div>
  );
};

Dashboard.propTypes = {
  handleLogout: PropTypes.func.isRequired,
};

export default Dashboard;
