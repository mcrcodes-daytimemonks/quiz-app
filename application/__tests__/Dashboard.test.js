import { render, screen } from "@testing-library/react";
import Dashboard from "../components/Dashboard";
import "@testing-library/jest-dom";

const props = {
  handleLogout: jest.fn(),
};

const renderDashboard = () => render(<Dashboard {...props} />);

describe("Dashboard", () => {
  it("matches snapshot", () => {
    const { asFragment } = renderDashboard();

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders a call to action", () => {
    renderDashboard();
    const callToAction = /please choose a category/i

    expect(screen.getByText(callToAction)).toBeInTheDocument();
  });
});
