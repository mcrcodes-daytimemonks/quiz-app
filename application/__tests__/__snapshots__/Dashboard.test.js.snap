// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Dashboard
    </h1>
    <div>
      <p>
        Welcome, 
        <span />
      </p>
      <p>
        Please choose a category
      </p>
      <button>
        JavaScript
      </button>
      <button>
        HTML
      </button>
      <button>
        mySQL
      </button>
      <label
        for="question-limit"
      >
        How many questions?
      </label>
      <input
        min="1"
        step="1"
        type="number"
        value="1"
      />
      <br />
      <button
        type="button"
      >
        Play
      </button>
      <br />
      <button
        type="button"
      >
        Logout
      </button>
    </div>
  </div>
</DocumentFragment>
`;
